src/main/java/uk/gov/hmcts/befta/data/CollectionVerificationConfig.java

```diff
@@ -51,4 +56,42 @@ public boolean isDefault() {
         return this == DEFAULT;
     }
 
+    public static CollectionVerificationConfig getVerificationConfigFrom(Collection<?> collection) {
+        if (collection == null || collection.isEmpty()) {
+            return DEFAULT;
+        }
+        Object firstElement = collection.iterator().next();
+        if (firstElement instanceof CollectionVerificationConfig) {
+            return (CollectionVerificationConfig) firstElement;
+        }
+        if (firstElement instanceof Map<?, ?>) {
+            Map<?, ?> firstMap = (Map<?, ?>) firstElement;
+            if (firstMap.containsKey(OPERATOR_FIELD_NAME) || firstMap.containsKey(ORDERING_FIELD_NAME)
+                    || firstMap.containsKey(ELEMENT_ID_FIELD_NAME)) {
+                CollectionVerificationConfig config = new CollectionVerificationConfig();
+                Operator operator = Operator.of((String) firstMap.get(OPERATOR_FIELD_NAME));
+                if (operator != null) {
+                    config.setOperator(operator);
+                }
+                Ordering ordering = Ordering.of((String) firstMap.get(ORDERING_FIELD_NAME));
+                if (ordering != null) {
+                    config.setOrdering(ordering);
+                } else {
+                    config.setOrdering(BeftaMain.getConfig().getDefaultCollectionAssertionMode());
+                }
+
+                if (config.getOrdering() == Ordering.UNORDERED) {
+                    String idString = (String) firstMap.get(ELEMENT_ID_FIELD_NAME);
+
+                    if (idString != null) {
+                        config.setElementId(idString);
+                    } else {
+                        config.setElementId(ElementIdFinder.findElementIds(collection));
```

Comment:
If element ID finder can't find a good ID candidate, then the test should fail with a test data exception.

Reply:
See line 89 in ElementIDFinder